Firestore/src/Query.php

```diff
@@ -1014,4 +978,144 @@ private function basePath()
+            throw new \InvalidArgumentException(sprintf(
+                'Value cannot be a `%s` value.',
+                FieldValue::class
+            ));
+        }
+
+        if (!($fieldPath instanceof FieldPath)) {
+            $fieldPath = FieldPath::fromString($fieldPath);
+        }
+
+        $escapedPathString = $fieldPath->pathString();
+
+        // alias for friendlier error message below.
+        $originalOperator = $operator;
+        $operator = array_key_exists(strtolower($operator), $this->shortOperators)
+            ? $this->shortOperators[strtolower($operator)]
+            : $operator;
+
+        try {
+            FieldFilterOperator::name($operator);
+        } catch (\UnexpectedValueException $e) {
+            throw new \InvalidArgumentException(sprintf(
+                'Operator %s is not a valid operator',
+                $originalOperator
+            ));
+        }
+
+        if ($escapedPathString === self::DOCUMENT_ID) {
+            if ($operator === FieldFilterOperator::IN) {
+                $value = array_map(function ($value) use ($basePath) {
+                    return $this->createDocumentReference($basePath, $value);
+                }, (array) $value);
+            } else {
+                $value = $this->createDocumentReference($basePath, $value);
+            }
+        }
+
+        if ((is_float($value) && is_nan($value)) || is_null($value)) {
+            if ($operator !== FieldFilterOperator::EQUAL) {
+                throw new \InvalidArgumentException('Null and NaN are allowed only with operator EQUALS.');
+            }
+
+            $unaryOperator = is_nan($value)
+                ? self::OP_NAN
+                : self::OP_NULL;
+
+            $filter = [
+                'unaryFilter' => [
+                    'field' => [
+                        'fieldPath' => $escapedPathString
+                    ],
+                    'op' => $unaryOperator
+                ]
+            ];
+        } else {
+            $encodedValue = $operator === FieldFilterOperator::IN
+                ? $this->valueMapper->encodeMultiValue((array)$value)
+                : $this->valueMapper->encodeValue($value);
+
+            $filter = [
+                'fieldFilter' => [
+                    'field' => [
+                        'fieldPath' => $escapedPathString,
+                    ],
+                    'op' => $operator,
+                    'value' => $encodedValue
+                ]
+            ];
+        }
+        
+        return $filter;
+    }
+
+    /**
+     * Find and set the the filter type to either Field or Unary.
+     *
+     * @param array $filter A field/unary filter array.
+     * @return array A non composite Filter array.
+     * @throws \InvalidArgumentException If an invalid filter is encountered.
+     */
+    private function findAndSetNonCompositeFilter($filter)
+    {
+        $nonCompositeFilter = $this->createNonCompositeFilter(
```

Comment:
rename to `$fieldFilter` and `createFieldFilter`

Reply:
Done!

Reply:
Also changing the name of the above function from `findAndSetNonCompositeFilter` to `findAndSetFilter`.